symengine/series_flint.cpp

```diff
@@ -2,6 +2,31 @@
 #include <symengine/series_visitor.h>
 
 #ifdef HAVE_SYMENGINE_FLINT
+
+#include <csetjmp>
+#include <csignal>
+#include <cstdlib>
+#include <iostream>
+
+jmp_buf env;
+
+void on_sigabrt (int signum)
+{
+    longjmp (env, 1);
+    throw std::runtime_error("Flint abort caught in series_flint.cpp");
+}
+
+void try_and_catch_abort (std::function< void ()> func)
+{
+    if (setjmp (env) == 0) {
+        signal(SIGABRT, &on_sigabrt);
+        func();
+    }
+    else {
+        std::cout << "aborted\n";
```

Comment:
You should restore the teuchos handler here and also throw the runtime_error from here